### DISCUSSION TITLE
On Windows Docker Desktop, it seems that an equivalent of --sig-proxy=false is required

### DISCUSSION THREAD

**Discussion by FlattusBlastus** [nextcloud/all-in-one#1942] [Questions] [answered]
I have been trying to make docker compose work. It seems that on Windows Docker Desktop an equivalent of --sig-proxy=false is required.

I get this error: Cannot connect to the docker socket. Cannot proceed.

Messing about with the environment variables hasn’t helped.

Begin docker-compose.yaml
version: '3.9'
volumes:
  nextcloud_aio_mastercontainer:
    name: nextcloud_aio_mastercontainer
  nextcloud_aio_nextcloud_datadir:
    name: nextcloud_aio_nextcloud_datadir
    driver: local
    driver_opts:
      o: bind
      type: none
      device: //e/ncdata
  nextcloud_aio_backupdir:
    name: nextcloud_aio_backupdir
    driver: local
    driver_opts:
      o: bind
      type: none
      device: //e/ncbackup
networks:
  nextcloud-aio:
    name: nextcloud-aio
services:
  nextcloud:
    image: nextcloud/all-in-one:latest
    restart: unless-stopped
    container_name: nextcloud-aio-mastercontainer
    volumes:
      - nextcloud_aio_mastercontainer:/mnt/docker-aio-config
      - /var/run/docker.sock:/var/run/docker.sock:ro
    ports:
      - 8080:8080
    environment:
      - APACHE_PORT=11000
      - APACHE_IP_BINDING=127.0.0.1
      - NEXTCLOUD_DATADIR="nextcloud_aio_nextcloud_datadir"
      - SKIP_DOMAIN_VALIDATION=true
      - NEXTCLOUD_MEMORY_LIMIT=2048M
    networks:
      - nextcloud-aio

**Comment by szaimen** [comment]
Hi, here is a working example for windows wirh caddy as reverse proxy: https://github.com/nextcloud/all-in-one/discussions/588#discussioncomment-4880429
Hopefully it helps you figuring things out.

As a sidenote: this doesnt look like a correct path: `device: //e/ncdata`

**Reply by FlattusBlastus** [reply]
Adding Caddy in doesn't help with the error: Cannot connect to the docker socket. Cannot proceed.

Also in Windows, thats drive E, ncdata folder.

My compose is probably as optimal as you can ask for on the Windows side but it doesn't work. The bug is likely with Docker Desktop and should be noted in the Windows implementation section. Something like:

Because Docker Compose does not support the --sig-proxy=false function, users using Windows Docker Desktop must use the docker run statement.

**Reply by szaimen** [reply]
Nope the sig-proxy is definitely not the problem.

See 
![Screenshot_20230206_171724_Brave](https://user-images.githubusercontent.com/42591237/217025763-c84ccada-5df5-48fe-a367-dce55a36e8cc.png)

**Comment by szaimen** [comment, marked as answer]
Did you already try `- //var/run/docker.sock:/var/run/docker.sock:ro`
instead of `- /var/run/docker.sock:/var/run/docker.sock:ro` ?

**Reply by FlattusBlastus** [reply]
Yes that worked!!!
Now the compose is saying that - NEXTCLOUD_DATADIR="nextcloud_aio_nextcloud_datadir" is wrong and that this should begin with a slash. However this env variable works in docker run.

**Reply by szaimen** [reply]
Ah yes, you need to remove the quotes.
`- NEXTCLOUD_DATADIR=nextcloud_aio_nextcloud_datadir` should work. However did you  already adjust the device as adviced below?

**Reply by FlattusBlastus** [reply]
Trying an experiment. Yes removing the quotes worked as well.

**Comment by szaimen** [comment]
Also, device for the volume should probably be rather `-o device="/host_mnt/e/ncdata`
instead of `-o device="//e/ncdata` as described here: https://github.com/nextcloud/all-in-one#how-to-change-the-default-location-of-nextclouds-datadir
Sane for the backup volume.

**Reply by FlattusBlastus** [reply]
/host_mnt/e/ncdata nor //e/ncdata or the backup is working in compose. /host_mnt/e/ncdata didnt work in docker run either but strangely the backup did. //e/ncdata and //e/ncbackup worked in docker run.

**Reply by szaimen** [reply]
Okay good then

**Reply by FlattusBlastus** [reply]
My updated docker compose... may it help you. The nextcloud datadir and backupdir variables are being ignored. Moving back to docker run. (Also, no dri device available in windows)
--
version: '3.9'

volumes:
  nextcloud_aio_mastercontainer:
    name: nextcloud_aio_mastercontainer
  nextcloud_aio_nextcloud_datadir:
    name: nextcloud_aio_nextcloud_datadir
    driver: local
    driver_opts:
      o: bind
      type: none
      device: /host_mnt/e/ncdata
  nextcloud_aio_backupdir:
    name: nextcloud_aio_backupdir
    driver: local
    driver_opts:
      o: bind
      type: none
      device: /host_mnt/e/ncbackup

networks:
  nextcloud-aio:
    name: nextcloud-aio

services:
  nextcloud:
    image: nextcloud/all-in-one:beta
    restart: unless-stopped
    container_name: nextcloud-aio-mastercontainer
    volumes:
      - nextcloud_aio_mastercontainer:/mnt/docker-aio-config
      - //var/run/docker.sock:/var/run/docker.sock:ro
    ports:
      - 8080:8080
    environment:
      - APACHE_PORT=11000
      - NEXTCLOUD_DATADIR=nextcloud_aio_nextcloud_datadir
      - SKIP_DOMAIN_VALIDATION=true
      - NEXTCLOUD_MEMORY_LIMIT=2048M
    networks:
      - nextcloud-aio

**Comment by FlattusBlastus** [comment]
One additional note: theres no such thing as network_mode: host in Windows Docker Desktop. Instead of using hostname:port in reverse proxy configs, use host.docker.internal:port

**Comment by FlattusBlastus** [comment]
This is working; i.e files getting to ncdata and ncbackup.
--
docker run ^
--sig-proxy=false ^
--name nextcloud-aio-mastercontainer ^
--restart always ^
--publish 8080:8080 ^
-e APACHE_PORT=11000 ^
-e NEXTCLOUD_DATADIR="nextcloud_aio_nextcloud_datadir" ^
-e SKIP_DOMAIN_VALIDATION=true ^
-e NEXTCLOUD_MEMORY_LIMIT=2048M ^
--volume nextcloud_aio_mastercontainer:/mnt/docker-aio-config ^
--volume //var/run/docker.sock:/var/run/docker.sock:ro ^
nextcloud/all-in-one:beta

docker volume create ^
--driver local ^
--name nextcloud_aio_nextcloud_datadir ^
-o device="//e/ncdata" ^
-o type="none" ^
-o o="bind"

docker volume create ^
--driver local ^
--name nextcloud_aio_backupdir ^
-o device="//e/ncbackup" ^
-o type="none" ^
-o o="bind"
